### PR TITLE
job-info: cleanup list attributes code and use %ju instead of %llu

### PR DESCRIPTION
**Pull request by chu11**
per issue #2884  and #2885, pretty minor stuff

### CODE REVIEW

File: src/modules/job-info/list.c [hunk 1]
```diff
@@ -488,25 +488,47 @@ void list_id_cb (flux_t *h, flux_msg_handler_t *mh,
...
-                           "nnodes",
-                           "ranks",
-                           "success") < 0) {
+    json_t *a = NULL;
+
+    if (!(a = json_array ())) {
+        errno = ENOMEM;
+        goto error;
+    }
+    if (list_attrs_append (a, "userid") < 0
```
Comment (garlick) [inline src/modules/job-info/list.c:515]: Suggestion: use an array so code doesn't change if attr list changes, and array might be reusable in other contexts?

E.g.
```c
static const char *attr_table[] = { "priority",  ..., "success", NULL };
...
int i;
for (i = 0; attr_table[i] != NULL; i++) {
    if (list_attrs_append (a, attr_table[i]) < 0)
        /* error */
}
```

Comment (chu11) [reply src/modules/job-info/list.c:515]: ahh good idea.

**Comment by chu11** [comment]
re-pushed, tweaking a bit per @garlick comments above

**Comment by grondo** [comment]
Thanks for the work!

**Comment by garlick** [comment]
Looks good, feel free to set MWP.

**Comment by chu11** [comment]
rebased on master, hoping that'll kick travis into doing what it needs to do

**Pull request merged**